scrape.py

```diff
@@ -17,22 +17,83 @@ def get_url(prefix, file_title, extension=None) :
     return return_url
 
 
-def get_response(url):
+def get_response(url, auth):
     request = Request(url, headers={'User-Agent': 'Mozilla/5.0'})
+
+    if auth:
+        request.add_header('Authorization', "token YOUR_TOKEN")
+
     response = urlopen(request)
 
     return response.read()
 
 
 def get_languages_url_from_json(json_data):
-    return json_data["payload"]["pull_request"]["head"]["repo"]["languages_url"]
+    try:
+        return json_data["payload"]["pull_request"]["head"]["repo"]["languages_url"]
+    except KeyError: # Likely missing "pull_request" key, but may be others in the json_data
+        #print("KeyError")
+        return None
+    except TypeError: # "None" or other non-json datatype within json_data
+        #print("TypeError")
+        return None
+    except:
+        print("Unhandled error!")
```

Comment:
Maybe we should also print out the Exception itself as well

Reply:
sounds good